src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3191,8 +3189,8 @@ jint Arguments::finalize_vm_init_args(bool patch_mod_javabase) {
     }
   }
 
-  if (UseSharedSpaces && patch_mod_javabase) {
-    no_shared_spaces("CDS is disabled when " JAVA_BASE_NAME " module is patched.");
+  if (UseSharedSpaces && _patch_mod_prefix != nullptr) {
+    no_shared_spaces("CDS is disabled when any module is patched.");
```

Comment:
But you are now disabling CDS if any module is patched, when the constraint AFAIK is only on the base module.

Reply:
Corrected to restore previous behavior and different messages between patching java.base and any other module.